File: cpp/1343.cpp
#include <iostream>
#include <vector>
#include <string>
#include <numeric>
using namespace std;

class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        threshold *= k;
        int cnt = accumulate(arr.begin(), arr.begin() + k, 0);
        int ans = cnt >= threshold ? 1 : 0;
        for(int i = k; i < arr.size(); ++i) {
            cnt = cnt - arr[i - k] + arr[i];
            if (cnt >= threshold) {
                ++ans;
            }
        }
        return ans;
    }
};

File: python/1343.py
class Solution(object):
    def numOfSubarrays(self, arr, k, threshold):
        """
        :type arr: List[int]
        :type k: int
        :type threshold: int
        :rtype: int
        """
        threshold *= k
        cnt = sum(arr[:k])
        ans = 1 if cnt >= threshold else 0
        for i in range(k, len(arr)):
            cnt = cnt - arr[i - k] + arr[i]
            if cnt >= threshold:
                ans += 1
        return ans